### DISCUSSION TITLE
[FIXED] Exception 28 on RTC NTP Server on ESP32 restart

### DISCUSSION THREAD

**Discussion by gfn256** [arendst/Tasmota#17338] [Q&A] [answered]
Whenever RtcNtpServer is ON, if the device reboots, (even through regular software restart), the i2c pins are reset in settings (to none), and the weblog shows:

00:00:00.003 HDW: ESP32-D0WDQ6
00:00:00.072 UFS: FlashFS mounted with 304 kB free
00:00:00.152 CFG: Loaded from File, Count 18
**00:00:00.156 FRC: Some settings have been reset (5)**
00:00:00.165 Project tasmota - Tasmota Version 12.1.1(TasmoCompiler-esp32generic)-2_0_4_1(2022-12-08T18:58:20)
00:00:00.020 WIF: Connecting to AP1 TP-LINK_5B69 Channel 11 BSSId 68:FF:7B:78:17:28 in mode 11n as tasmota-E3A3B4-0948...
00:00:03.865 WIF: Connected
00:00:04.122 HTP: Web server active on tasmota-XXXXXX-XXXX with IP address 192.168.1.136
20:22:45.167 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.1.1(TasmoCompiler-esp32generic)","FallbackTopic":"cmnd/DVES_XXXXXX_fb/","GroupTopic":"cmnd/tasmotas/"}}
20:22:45.180 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-XXXXXX-XXXX","IPAddress":"192.168.1.136"}}
20:22:45.197 RSL: INFO3 = {"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC":"4014d34f","EXCVADDR":"00000000","CallChain":["4014d34c","4013d920","4014e245","4014e2a3","4013d131","400e81a6","400e827f","400fc455","400fd481","40127d57","400e48d1"]},"BootCount":7}}
20:22:50.886 RSL: STATE = {"Time":"2022-12-08T20:22:50","Uptime":"0T00:00:12","UptimeSec":12,"Heap":159,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":39,"MqttCount":0,"Wifi":{"AP":1,"SSId":"XXXXXXXXXXXXXXXX","BSSId":"XXXXXXXXXXXXXX","Channel":11,"Mode":"11n","RSSI":42,"Signal":-79,"LinkCount":1,"Downtime":"0T00:00:05"}}
20:22:50.911 RSL: SENSOR = {"Time":"2022-12-08T20:22:50","ESP32":{"Temperature":31.1},"TempUnit":"C"}

However if I turn RtcNtpServer OFF, this does not happen. Also if I remove RTC module before the restart this does not happen.

**Comment by sfromis** [comment]
Since there is the number (5) after the message about the reset, that indicates a boot loop. Tasmota has a recovery feature to disable setup in steps till able to boot without crashing.
`"RestartReason":{"Exception":28,"Reason":"LoadProhibited"` ....

**Comment by arendst** [comment]
Issue verified as exception on ESP32 (On ESP8266 it works fine). Will investigate....

**Comment by arendst** [comment, marked as answer]
Fixed in latest dev branch

**Reply by gfn256** [reply]
Thanks! How can I use the latest dev branch in TasmoCompiler?

**Reply by sfromis** [reply]
You should have branch development selected instead of release.

Dont remember details of how to use that tool, but you should be able to do that selection using this:
https://gitpod.io/#https://github.com/benzino77/tasmocompiler/tree/development

**Reply by gfn256** [reply]
Thanks so much! Yes I managed with your link and rtcntpserver works just as I wanted. No more resets on restart, and NTPserver is reachable on my network. Many thanks to you!  One thing doesn't seem to work with this version is ping. I try pinging, example:  ping4 192.168.1.1 and all I get is
 RESULT = {"Ping":{"192.168.1.108":{"Reachable":**false**,"IP":"192.168.1.108","Success":**0**,"Timeout":4,"MinTime":0,"MaxTime":0,"AvgTime":0}}}
I've tried any known addresses that work from my other tasmotas, but on this NTPserver tasmota I only get above result? Something to do with development branch?
